geode-core/src/distributedTest/java/org/apache/geode/internal/jta/dunit/TxnTimeOutDUnitTest.java

```diff
@@ -205,8 +207,11 @@ public static void runTest1() throws Exception {
       Context ctx = cache.getJNDIContext();
       UserTransaction utx = (UserTransaction) ctx.lookup("java:/UserTransaction");
       utx.begin();
+      assertThat(utx.getStatus() == Status.STATUS_ACTIVE);
       utx.setTransactionTimeout(2);
-      Thread.sleep(6000);
+      long start = System.currentTimeMillis();
+      Thread.sleep(2000);
+      waitUntilTransactionTimeout(utx, start);
```

Comment:
Can it be replaced using GeodeAwaitility and remove all the sleeps...
GeodeAwaitility.await().until(() -> utx.getStatus() != Status.STATUS_NO_TRANSACTION);

Reply:
No, it checks the thread that begins the transaction. GeodeAwaitility is another thread and it won't work.

Reply:
PollInSameThread allows to wait in same thread.

GeodeAwaitility.await().pollInSameThread().until(() -> {
        return utx.getStatus() == Status.STATUS_NO_TRANSACTION;
      });